File: src/main/java/ac/kr/deu/connect/luck/food_truck/FoodTruckReviewRequest.java
package ac.kr.deu.connect.luck.food_truck;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "리뷰 생성 요청 DTO")
public record FoodTruckReviewRequest(
        @Schema(description = "리뷰 내용", example = "맛있어요")
        String content,
        @Schema(description = "이미지 URL", example = "https://picsum.photos/1600/900")
        String imageUrl,
        @Schema(description = "평점", example = "5")
        int score,
        @Schema(description = "푸드트럭 ID", example = "1")
        Long foodTruckId,
        @Schema(description = "작성자 ID", example = "1")
        Long authorId
) {
}

File: src/main/java/ac/kr/deu/connect/luck/foodtruck/controller/request/FoodTruckReviewRequest.kt
package ac.kr.deu.connect.luck.foodtruck.controller.request

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import org.springframework.web.multipart.MultipartFile

@Schema(description = "푸드트럭 리뷰 등록 요청 DTO")
data class FoodTruckReviewRequest(
    @field:Schema(description = "리뷰 내용", example = "맛있어요")
    val content: String,
    @field:Schema(description = "평점 1~5 사이값만 입력하십시오", example = "5")
    @field:Min(1) @field:Max(5)
    val rating: Int,
    @field:Schema(description = "이미지", example = "image.jpg")
    val image: MultipartFile?
) {
    override fun toString(): String {
        return "FoodTruckReviewRequest(content='$content', rating=$rating image=${image?.originalFilename})"
    }
}
